### PR TITLE
Demo browser: sort by length (closed #1739)

### LINKED ISSUE #1739: Allow sorting demos by date/size/..

### PR DESCRIPTION
**Pull request by ChillerDragon**
Full credits go to @def- code was taken from this ddnet commit
https://github.com/ddnet/ddnet/commit/f9d9fee31448d0d98b35456f63931a387208e6a4

teeworlds_demo_sort_preview

### CODE REVIEW

**Comment by ChillerDragon** [comment]
It would be nice to also show the lengths in the list and add a option to filter by markers.

File: src/game/client/components/menus_demo.cpp [hunk 1]
```diff
@@ -401,6 +402,18 @@ void CMenus::DemolistOnUpdate(bool Reset)
 	m_DemolistSelectedIsDir = m_DemolistSelectedIndex < 0 ? false : m_lDemos[m_DemolistSelectedIndex].m_IsDir;
 }
 
+bool CMenus::FetchHeader(CDemoItem &Item)
```
Comment (oy) [inline src/game/client/components/menus_demo.cpp:405]: You should pass by pointer https://github.com/teeworlds/documentation/blob/master/nomenclature.md#passing-variables

**Comment by oy** [comment]
As there's a new `length` column it might be best to show the length of each demo in the list.

**Comment by ChillerDragon** [comment]
demo_cols

**Pull request merged**